clippy_lints/src/utils/internal_lints/metadata_collector.rs

```diff
@@ -485,16 +486,44 @@ fn extract_attr_docs_or_lint(cx: &LateContext<'_>, item: &Item<'_>) -> Option<St
 ///
 /// Would result in `Hello world!\n=^.^=\n`
 fn extract_attr_docs(cx: &LateContext<'_>, item: &Item<'_>) -> Option<String> {
-    cx.tcx
+    let mut lines = cx
+        .tcx
         .hir()
         .attrs(item.hir_id())
         .iter()
-        .filter_map(|x| x.doc_str().map(|sym| sym.as_str().to_string()))
-        .reduce(|mut acc, sym| {
-            acc.push_str(&sym);
-            acc.push('\n');
-            acc
-        })
+        .filter_map(ast::Attribute::doc_str);
+    let mut docs = String::from(&*lines.next()?.as_str());
+    let mut in_code_block = false;
+    for line in lines {
+        docs.push('\n');
+        let line = line.as_str();
+        let line = &*line;
+        if let Some(info) = line.trim_start().strip_prefix("```") {
+            in_code_block = !in_code_block;
+            if in_code_block {
+                let mut parts = info
+                    .trim()
+                    .split(',')
+                    // remove rustdoc directives
+                    .filter(|&s| !matches!(s, "" | "ignore" | "no_run" | "should_panic"));
+                docs.push_str("```");
+                match parts.next() {
+                    // if no language is present, fill in "rust"
+                    None => docs.push_str("rust"),
+                    Some(first) => {
+                        docs.push_str(first);
+                        for part in parts {
+                            docs.push(',');
+                            docs.push_str(part);
+                        }
```

Comment:
Are there actually any additional parts we want to include here? The website uses highlight.js which can only take one language and no additional parameters to my understanding. The best solution would probably to only keep the main language and drop the rest, unless someone knows of a case where we need this.

I actually added some extra handling in the website code that does the same [here](https://github.com/rust-lang/rust-clippy/pull/7279/files#diff-c618f97d2a35c1d819a1df7a09772499726a909c94959c8f6a4e5c5d7230ed0cR374-R381). Having it in the collector itself is something I didn't think of. Nicely done.

Reply:
Yeah dropping the rest would be fine. I waffled on this. YAGNI.

Oh, the JS solution is nice too! I don't have a strong preference for one or the other. But we should just keep one. I'll let you decide which.

Reply:
Good, we can merge this once the additional parts are gone.

Having it here makes more sense IMO. The other solution runs for each block on each page load, that's a bit excessive in comparison. And you added the nice part that no language means `rust`. I've also touched this method to adjust some markdown for the website in that PR, so it's the correct place I would say :upside_down_face:.